File: DotNet/UITestFramework.Tests/actions/CommonActions.cs
using OpenQA.Selenium;

namespace UITestFramework.Tests.actions
{
    public sealed class CommonActions
    {
        private readonly IWebDriver driver;

        public CommonActions(IWebDriver driver)
        {
            this.driver = driver;
        }

        internal void ClickElement(By by) => driver.FindElement(by).Click();

        internal bool ElementDisplayed(By by)
        {
            try
            {
                driver.FindElement(by);
                return true;
            }
            catch
            {
                return false;
            }
        }

        internal string GetTextFromElement(By by) => driver.FindElement(by).Text;
    }
}

File: Java/src/main/java/webdriverstructure/actions/CommonActions.java
package webdriverstructure.actions;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class CommonActions extends ActionBase{
    public CommonActions(WebDriver driver) {
        super(driver);
    }

    public void ClickElement(By by){
        driver.findElement(by).click();
    }

    public void EnterText(By by, String text){
        WebElement element = driver.findElement(by);
        element.clear();
        element.sendKeys(text);
    }

    public String GetTextFromElement(By by){
        return driver.findElement(by).getText();
    }

    public boolean ElementDisplayed(By by){
        try{
            driver.findElement(by);
            return true;
        } catch(Exception ex) {
            return false;
        }
    }

    public int GetNumberOfElementsMatching(By by){
        return driver.findElements(by).size();
    }
}
